### ISSUE TITLE
[Feature Requst] Show Progress

### ISSUE THREAD

**Issue by lixin-wei** [flamegraph-rs/flamegraph#270]
In my case, `perf.data` is huge(10GB+) and usually takes 30+ min to generate the flame graph.

It will be much more friendly if there is a progress bar.

**Comment by djc** [comment]
Unfortunately I don't have much time for maintaining flamegraph. Maybe consider submitting a PR?

**Comment by djc** [comment]
No, but it's pretty simple. We get the inferno crate to collapse the data at https://github.com/flamegraph-rs/flamegraph/blob/main/src/lib.rs#L367 and generate the flamegraph at https://github.com/flamegraph-rs/flamegraph/blob/main/src/lib.rs#L427. You should figure which of those is slow. Then, I think you might want to wrap an indicatif thing around the input to either of those things.

**Comment by smoelius** [comment]
@djc Do these look like the right links? I'm speculating based on the date of your comment:
- https://github.com/flamegraph-rs/flamegraph/blob/1aeb2a82cbe228d715f38751fd7e1a5c3586dfd4/src/lib.rs#L367
- https://github.com/flamegraph-rs/flamegraph/blob/1aeb2a82cbe228d715f38751fd7e1a5c3586dfd4/src/lib.rs#L427

**Comment by smoelius** [comment]
Also, for me, the bottleneck appears to be `perf script`: https://github.com/flamegraph-rs/flamegraph/blob/1aeb2a82cbe228d715f38751fd7e1a5c3586dfd4/src/lib.rs#L128

Is this surprising to you? (I don't know what this command does.)

**Comment by gth828r** [comment]
This is a problem which I had, so I can probably take a stab at adding this feature. Implementing a true progress bar with perf script is not obvious to me, but I think a spinner with a message about what is running and an elapsed time is a good start, and that is fairly simple to add.

**Comment by gth828r** [comment]
Ok, after a few different iterations, I found something that I like and it is available as a PR at https://github.com/flamegraph-rs/flamegraph/pull/343
